src/emulator/storage/files.ts

```diff
@@ -585,18 +587,22 @@ export class StorageLayer {
         continue;
       }
 
-      let decodedBlobPath = decodeURIComponent(blobPath);
-      const decodedBlobPathSep = getPathSep(decodedBlobPath);
+      let fileName = metadata.name;
+      const objectNameSep = getPathSep(fileName);
       // Replace all file separators with that of current platform for compatibility
-      if (decodedBlobPathSep !== path.sep) {
-        decodedBlobPath = decodedBlobPath.split(decodedBlobPathSep).join(path.sep);
+      if (fileName !== path.sep) {
+        fileName = fileName.split(objectNameSep).join(path.sep);
       }
 
+      const filepath = this.path(metadata.bucket, fileName);
+
+      const decodedBlobPath = decodeURIComponent(blobPath);
       const blobDiskPath = this._persistence.getDiskPath(decodedBlobPath);
 
       const file = new StoredFile(metadata, blobDiskPath);
-      this._files.set(decodedBlobPath, file);
+      this._files.set(filepath, file);
 
+      this._persistence.addDiskPathMapping(filepath, decodedBlobPath);
```

Comment:
If I'm understanding this correctly, new files added to the emulator are stored on disk as uuids, imported files maintain their existing formatting. IMO let's coalesce all files, existing or new, to use the new randomly generated ID system as opposed to maintaining both. 

This will also have the added benefit of not needing to expose `addDiskPathMapping` from `Persistence`

Reply:
Right now the imported files are copied into the same unique id format as the new files created in the emulator, but I needed the addDiskPathMapping to add the created ids into the map. 
I get what you're saying though and I see why we wouldn't want to expose addDiskPathMapping so I'll try to bake the mapping logic into the persistence layer itself to maintain a cleaner interface.